lib/index.ts

```diff
@@ -0,0 +1,134 @@
+import Controller from './controller'
+import Partial from './partial'
+import { domReady } from './utils'
+
+enum Data {
+  Controller = '[data-controller]',
+  Partial = '[data-partial]',
+}
+
+enum DataAttribute {
+  Controller = 'controller',
+  Partial = 'partial',
+}
+
+type Callback = (el: Element) => void
+
+type Cache = {
+  [DataAttribute.Controller]: Element[]
+  [DataAttribute.Partial]: Element[]
+}
+
+export class Embroidery {
+  private cache: Cache = {
```

Comment:
Use readonly when possible

```suggestion
  private readonly cache: Cache = {
```